libraries/core_libs/consensus/src/final_chain/rewards_stats.cpp

```diff
@@ -0,0 +1,75 @@
+#include "final_chain/rewards_stats.hpp"
+
+namespace taraxa {
+
+bool RewardsStats::addTransaction(const trx_hash_t& tx_hash, const addr_t& validator) {
+  auto found_tx = txs_validators_.find(tx_hash);
+
+  // Already processed tx
+  if (found_tx != txs_validators_.end()) {
+    return false;
```

Comment:
maybe there should be assert as all trxs needs to be unique in syncblock, or not here but in processblock should be assert on return value of this function

Reply:
ah this must be without assert, it is actually processing also duplicate txs(from dag blocks -> txs hashes):
```
for (const auto& dag_block : sync_blk.dag_blocks) {
    const addr_t& dag_block_author = dag_block.getSender();
    for (const auto& tx_hash : dag_block.getTrxs()) {
      addTransaction(tx_hash, dag_block_author);
    }
  }
```